src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -515,13 +516,7 @@ public long scheduleTimeout(ContextInternal context, Handler<Long> handler, long
   }
 
   public AbstractContext getContext() {
-    AbstractContext context = (AbstractContext) ContextInternal.current();
-    if (context != null && context.owner() == this) {
-      return context;
-    } else {
-      WeakReference<AbstractContext> ref = stickyContext.get();
-      return ref != null ? ref.get() : null;
-    }
+    return (AbstractContext) contextManager.getContext(this);
```

Comment:
So actually the ContextInternal.current() should remain here (as it does not cost an extra call) and the else statement should be in the context manager, sorry to not have been clear enough

Reply:
In other words, the call should only be called when we are not on the current thread.